TESTS/mbed_hal/spm/main.cpp

```diff
@@ -84,7 +84,7 @@ __attribute__((naked)) void call_mem(uint32_t addr)
     // since exception will be generated for invalid memory access.
     // Other instructions are for calling do_nothing function according to AAPCS.
     __ASM(
-        "LDR     r1, [r0]\n"
+        "LDR     r4, [r0]\n"
```

Comment:
I'm not sure why r1 is not backed up by compiler in `test_memory()` before calling to `call_mem()` - perhaps it is a bug in GCC_ARM.
But you cannot use `r4` as it will violate AAPCS standard. Please use `r3` instead. 
In addition also open ticket for GCC_ARM.

```suggestion
        "LDR     r3, [r0]\n"
```

Reply:
Since the compiler sees the definition of call_mem() when compiling test_memory(), it knows that r1 isn't used and doesn't need to be backed up. It would probably back it up if r1 was in the clobber list of the asm, as it should be.